### PR TITLE
Nikolai/feature/search

### PR DESCRIPTION
**Pull request by lizard-52**
Re implement the search function. Searches are case insensitive and will match any item with the search string contained in the name/make/description field.

Sorting works too.

### CODE REVIEW

File: code/app/src/main/java/com/example/sigma_blue/entity/item/ItemList.java [hunk 1]
```diff
@@ -23,16 +23,10 @@
 
 public class ItemList extends AEntityList<Item> {
     /* Attributes */
-    private ViewListModes displayMode;
+    private VisibleItemList visibleItemList;
+    private ArrayList<Item> visibleItemArrayList;
 
 
-    public enum ViewListModes {
```
Comment (waridh) [inline code/app/src/main/java/com/example/sigma_blue/entity/item/ItemList.java:29]: Whoops. I definitely left that there accidentally.

**Review by waridh — APPROVED** [review]
Its a good refactor. I should have moved away from enum to class sooner

**Comment by waridh** [comment]
Actually, why did we remove the unsorted state? It was implicitly sort by oldest entry.

**Review by waridh — CHANGES_REQUESTED** [review]
Fix the issue related to date queries please. I have outlined how. You can blame me all you want.

**Comment by lizard-52** [comment]
> Actually, why did we remove the unsorted state? It was implicitly sort by oldest entry.

I figured an 'unsorted' state doesn't really make sense since the list is always sorted. It's set up to be easy to change the default, you just need to change the constructor in ItemSortComparator.

File: code/app/src/main/java/com/example/sigma_blue/fragments/QueryFragment.java [hunk 2]
```diff
@@ -174,49 +166,61 @@ public void flipAscendBox(boolean p) {
...
+            } else {
+                // from the last week by default
+                start.add(Calendar.DATE, -5);
+            }
 
-            dateCheckBoxController(this.dateFilterBox);
-        }
+            startDatePicker.updateDate(
+                    start.get(Calendar.YEAR),
+                    start.get(Calendar.MONTH) - 1,
```
[the remark this answers is not in the document]
Comment (lizard-52) [reply code/app/src/main/java/com/example/sigma_blue/fragments/QueryFragment.java]: Fixed now

**Comment by waridh** [comment]
So it looks like date query is not fully fixed. The UI restoration is correct, but the search is wrong. Also, when I had the wrong date query selected and then selected reset query, the app crashes.

**Comment by waridh** [comment]
> So it looks like date query is not fully fixed. The UI restoration is correct, but the search is wrong. Also, when I had the wrong date query selected and then selected reset query, the app crashes.

To be specific, the ui will now start in the correct month, but we would still need to query in october to get november items. The app crashing on reset button one seems new though.

**Review by waridh — APPROVED** [review]
Working now. I want to ask about the default date range selection of 5 days into the past, but that's more ux than backend.

**Review by bpaul2 — APPROVED** [review]

**Pull request merged**